homeassistant/helpers/device_registry.py

```diff
@@ -351,6 +351,10 @@ class DeviceEntry:
     via_device_id: str | None = attr.ib(default=None)
     # This value is not stored, just used to keep track of events to fire.
     is_new: bool = attr.ib(default=False)
+    # This value is not stored, just used to keep track of whether the entry
+    # was restored from storage. Marked as private because it should only be used
+    # by the DeviceRegistry.
+    _is_restored: bool = attr.ib(default=False)
```

Comment:
since we use the term "restored" for entities state which get restored from storage, and this is for deleted devices that get restored, maybe a different name to avoid mixing up the two concepts? 
not sure about the name. `is_deleted_restored`?